CONTRIBUTING.md

```diff
@@ -39,11 +39,15 @@ You can find all the details in the [Contributing via Git](http://wiki.eclipse.o
 
 ## Making your Changes
 
+1. Consider to start with creating an issue on GitHub to discuss your plans and get the proper startup information.
 1. Fork the repository on GitHub
-1. Create a new branch for your changes
-1. Make your changes
+1. Create a new branch for your changes based on the 2.0.x branch. 
+   Please note: work based on other branches without prior discussion in an issue, may be in vain.
+1. If you use the eclipse IDE, please import our prefer formatter `eclipse-formatter-profile.xml` from the californium parent-folder and apply it to your changes (only :-) ).
+1. Make your changes 
 1. Make sure you include test cases for non-trivial features/changes
 1. Make sure the test suite runs successfully after you made your changes
+1. If new files are created, provide a proper license header
```

Comment:
we should also provide some guidance as to how the license header should look like.
Maybe something like:
`If new files are created, provide a proper license header. For that purpose, copy the license header from an existing source file and adapt it to reflect the current year and your organization/name. Please also make sure that the list of contributors only contains a single entry (representing your initial work on the file) and does not contain the entries from the file that you copied the header from.`

Reply:
Maybe we provide a file with just such a header.